processes/ingest/publisher_backlist.py

```diff
@@ -1,50 +1,50 @@
 import os
 from services import PublisherBacklistService
 
-from ..core import CoreProcess
 from logger import create_log
-from managers import S3Manager
+from managers import DBManager, S3Manager
+from ..record_buffer import RecordBuffer
+from .. import utils
 
 logger = create_log(__name__)
 
-class PublisherBacklistProcess(CoreProcess):
+class PublisherBacklistProcess():
     def __init__(self, *args):
-        super(PublisherBacklistProcess, self).__init__(*args[:4])
+        self.process = args[1]
+        self.ingest_period = args[2]
 
         self.limit = (len(args) >= 5 and args[4] and int(args[4]) <= 100) or None
         self.offset = (len(args) >= 6 and args[5]) or None
 
+        self.db_manager = DBManager()
+        self.db_manager.createSession()
+
+        self.record_buffer = RecordBuffer(db_manager=self.db_manager)
+
         self.s3_bucket = os.environ['FILE_BUCKET']
         self.s3_manager = S3Manager()
 
         self.publisher_backlist_service = PublisherBacklistService()
         
     def runProcess(self):
         try:
-            self.generateEngine()
-            self.createSession()
-
+            start_datetime = utils.get_start_datetime(process_type=self.process, ingest_period=self.ingest_period)
             self.publisher_backlist_service.delete_records()
 
-            if self.process == 'daily':
-                records = self.publisher_backlist_service.get_records(offset=self.offset, limit=self.limit)
-            elif self.process == 'complete':
-                records = self.publisher_backlist_service.get_records(full_import=True)
-            elif self.process == 'custom':
-                records = self.publisher_backlist_service.get_records(start_timestamp=self.ingestPeriod, offset=self.offset, limit=self.limit)
-            else: 
-                logger.warning(f'Unknown Publisher Backlist ingestion process type {self.process}')
-                return
-            for record in records:
-                self.addDCDWToUpdateList(record)
+            records = self.publisher_backlist_service.get_records(
+                start_timestamp=start_datetime,
+                offset=self.offset,
+                limit=self.limit
+            )
             
-            self.saveRecords()
-            self.commitChanges()
+            for record_mapping in records:
+                self.record_buffer.add(record_mapping.record)
+            
+            self.record_buffer.flush()
 
             logger.info(f'Ingested {len(self.records)} Publisher Backlist records')
```

Comment:
This can be updated to use record_buffer.ingest_count

Reply:
oops !